core/services/log/eth_subscriber.go

```diff
@@ -45,39 +45,42 @@ func (sub *ethSubscriber) backfillLogs(fromBlockOverride null.Int64, addresses [
 	ctxParent, cancel := utils.ContextFromChan(sub.chStop)
 	defer cancel()
 
+	var latestHeight int64 = -1
 	retryCount := 0
 	utils.RetryWithBackoff(ctxParent, func() (retry bool) {
 		if retryCount > 3 {
 			return false
 		}
 		retryCount++
 
-		ctx, cancel := context.WithTimeout(ctxParent, 10*time.Second)
-		defer cancel()
+		if latestHeight < 0 {
+			ctx, cancel := context.WithTimeout(ctxParent, 10*time.Second)
+			defer cancel()
 
-		latestBlock, err := sub.ethClient.HeadByNumber(ctx, nil)
-		if err != nil {
-			logger.Errorw("LogBroadcaster: Backfill - could not fetch latest block header, will retry", "err", err)
-			return true
-		} else if latestBlock == nil {
-			logger.Warn("LogBroadcaster: Got nil block header, will retry")
-			return true
+			latestBlock, err := sub.ethClient.HeadByNumber(ctx, nil)
```

Comment:
Do we have to use this? Can we ask the head tracker instead?

Reply:
hmm yes i think we can now, as dependencies handling were improved some time ago. I'll create a separate ticket